File: tasks/b5-nodejs-greenfield-api-or-cli/src/schemas.ts
import { z } from "zod";
import { parseAmount } from "./money.js";

export const transactionCreateSchema = z.object({
  amount: z.union([z.string(), z.number()]).transform((value, ctx) => {
    const parsed = parseAmount(value);
    if (!parsed) {
      ctx.addIssue({ code: z.ZodIssueCode.custom, message: "Amount must be greater than 0" });
      return z.NEVER;
    }
    return parsed;
  }),
  type: z.enum(["credit", "debit"]),
  description: z.string().max(200).optional(),
});

export type TransactionCreate = z.infer<typeof transactionCreateSchema>;

export interface TransactionResponse {
  id: number;
  amount: string;
  type: "credit" | "debit";
  description?: string;
}

export interface BalanceResponse {
  balance: string;
  currency: "USD";
}

File: tasks/a2-execute-two-parallel-worktrees/sandbox/expense-tracker/app/schemas.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class TransactionCreate(BaseModel):
    amount: float = Field(gt=0)
    category: str
    description: Optional[str] = None


class TransactionResponse(BaseModel):
    id: int
    amount: float
    category: str
    description: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class BalanceResponse(BaseModel):
    balance: float
